src/main/java/de/bonndan/nivio/input/Processor.java

```diff
@@ -0,0 +1,20 @@
+package de.bonndan.nivio.input;
+
+import de.bonndan.nivio.input.dto.LandscapeDescription;
+import de.bonndan.nivio.model.Landscape;
+
+/**
+ * Modifies the target landscape using input.
+ *
+ *
+ */
+abstract class Processor {
+
+    protected final ProcessLog processLog;
+
+    protected Processor(ProcessLog processLog) {
+        this.processLog = processLog;
+    }
```

Comment:
Do you use  an abstract class instead of an interface only because of the `ProcessLog` field?

Reply:
Yes, so far. We can change it to an interface and add the processlog as further argument to process/resolve.

Reply:
It's ok, I dont mind the abstract class. I think the only thing that "bothers" me is that I don't fully understand what the resolvers and the processors do and why - see comment above :)

Reply:
Until a few days ago I was not aware of that, too. I just observed that some of the classes in Indexer only worked on the input objects, while others only/mostly worked on the output.